# Task 31 — High-Usage Customer Share by Subscription Plan

## Business Question

What percentage of customers within each subscription plan are high-usage customers?

## Analysis

Calculated the total number of customers and the number of high-usage customers (gb_downloaded > 3000) for each subscription plan. The results were combined using merge(), and the percentage of high-usage customers was calculated for each plan.

## Code

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
users_by_plan_and_more_3000gb = (
    starlink[starlink["gb_downloaded"] > 3000]
    .groupby("plan_name").agg(
        high_users_amount = ("unique_id","count")
    )
).sort_values(
    "high_users_amount",
    ascending = False
)
total_customers = (
    starlink
    .groupby("plan_name")
    .agg(
        total_customers = ("unique_id", "count")
    )
)
result = total_customers.merge(
    users_by_plan_and_more_3000gb,
    left_on = "plan_name",
    right_on = "plan_name",
    how = "left"
)
result["high_usage_percent"] = ((result["high_users_amount"] / result["total_customers"]) * 100.0).round(2)
print(result)

             total_customers  high_users_amount  high_usage_percent
plan_name                                                          
Business                1019                146               14.33
Mini                     944                130               13.77
Residential             5959                793               13.31
Roam                    2078                285               13.72


### Business Insight

The share of high-usage customers is relatively similar across all subscription plans, ranging from 13.31% to 14.33%. Business has the highest proportion of high-usage customers (14.33%), while Residential has the lowest (13.31%). This suggests that subscription plan type has only a small difference in the proportion of high-usage customers in this dataset.